### PR TITLE
Docs for useLandmark

### PR DESCRIPTION
**Pull request by reidbarber**
Docs for the useLandmark hook.

Also, maybe we can consider creating some simple landmark components in RAC.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-aria/landmark/docs/useLandmark.mdx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+## Features
+
+Landmarks provide a way to designate important subsections of a page. They allow screen reader users to get an overview of the various sections of the page, and jump to a specific section.
+By default, browsers do not provide a consistent way to navigate between landmarks using the keyboard.
+The `useLandmark` hook enables keyboard navigation between landmarks, and provides a consistent experience across browsers.
+
+* <Keyboard>F6</Keyboard> and <Keyboard>Shift+F6</Keyboard> key navigation between landmarks
+* <Keyboard>Alt+F6</Keyboard> key navigation to the main landmark
+* Support for navigating nested landmarks
+* Support for navigating across iframes
```
Comment (snowystinger) [inline packages/@react-aria/landmark/docs/useLandmark.mdx:47]: doesn't this require extra work? what is that extra work?

Comment (reidbarber) [reply packages/@react-aria/landmark/docs/useLandmark.mdx:47]: A real iframe example may be difficult in our docs, so I could add a dummy example that doesn't render anything.

Comment (snowystinger) [reply packages/@react-aria/landmark/docs/useLandmark.mdx:47]: I think if we know what work is needed for support within iframes, we can just write about it, no need to create a full blown example. I just thought @devongovett had to do some extra work here when he implemented it in another project.

Comment (reidbarber) [reply packages/@react-aria/landmark/docs/useLandmark.mdx]: Ended up removing it since it was a bit complex.

**Review by snowystinger — APPROVED** [review]
I'm glad we didn't put a 'main' on in here, that'd make it harder for an actual screen reader user to navigate this page.
I kind of wish that we could do this with the page and have a non-rendering code sample here.

What do you think about linking to useToast docs? https://reactspectrum.blob.core.windows.net/reactspectrum/fdfe9b48964989250367507df653ac0395df3ce3/docs/react-aria/useToast.html since useToastContainer makes use of it?

**Review by snowystinger — DISMISSED** [review]
Not 100% that this is "navigation", that section is more about swapping content in/out.
Feels more like utility based on our current categories?

File: packages/@react-aria/landmark/docs/useLandmark.mdx [hunk 2]
```diff
@@ -0,0 +1,111 @@
+{/* Copyright 2020 Adobe. All rights reserved.
```
Comment (ktabors) [inline packages/@react-aria/landmark/docs/useLandmark.mdx:1]:
```suggestion
{/* Copyright 2023 Adobe. All rights reserved.
```

**Review by ktabors — APPROVED** [review]
Great!

Only concern is this page points out that we aren't using Landmarks in your docs and maybe we should be.

**Review by ktabors — DISMISSED** [review]

**Review by snowystinger — APPROVED** [review]

**Review by snowystinger — APPROVED** [review]

**Review by ktabors — APPROVED** [review]

**Pull request merged**